sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/WriteGroupedRowsToFiles.java

```diff
@@ -75,28 +84,35 @@ private org.apache.iceberg.catalog.Catalog getCatalog() {
       return catalog;
     }
 
-    private RecordWriter createWriter(IcebergDestination destination) throws IOException {
-      return new RecordWriter(getCatalog(), destination, "-" + UUID.randomUUID());
-    }
-
     @ProcessElement
     public void processElement(
-        ProcessContext c, @Element KV<ShardedKey<Row>, Iterable<Row>> element) throws Exception {
+        ProcessContext c,
+        @Element KV<ShardedKey<Row>, Iterable<Row>> element,
+        BoundedWindow window,
+        PaneInfo pane)
+        throws Exception {
 
       Row destMetadata = element.getKey().getKey();
       IcebergDestination destination = dynamicDestinations.instantiateDestination(destMetadata);
-      RecordWriter writer = createWriter(destination);
+      WindowedValue<IcebergDestination> windowedDestination =
+          WindowedValue.of(destination, window.maxTimestamp(), window, pane);
+      RecordWriterManager writer =
+          new RecordWriterManager(getCatalog(), filePrefix, maxFileSize, Integer.MAX_VALUE);
```

Comment:
With above change to autocloseable and rename, everything up to `close()` becomes:

```
try(PartitionedWriter writer = new PartitionedWriter(getCatalog(), filePrefix, maxFileSize, Integer.MAX_VALUE) {
   ...
}
```

Otherwise, you want a `try { ... } finally { close() }` pattern I think. But try-with-resources is easier to read and more robust.

Reply:
Done